addons/mail/models/mail_template.py

```diff
@@ -372,6 +372,33 @@ def render_template(self, template_txt, model, res_ids, post_process=False):
         }
         for res_id, record in res_to_rec.iteritems():
             variables['object'] = record
+            # object company dependent user signatures
+            object_create_user_signature = False
+            object_user_signature = False
+            if 'user_id' in self.env[model]._fields:
+                object_user_signature = record.user_id and \
+                    record.user_id.signature or False
+            if 'create_uid' in self.env[model]._fields:
+                object_create_user_signature = record.create_uid and \
+                    record.create_uid.signature or False
+            current_user_signature = self.env.user.signature or False
+            if 'company_id' in self.env[model]._fields and record.company_id:
+                res_users_obj_sudo = self.env['res.users'].sudo().with_context(
+                    force_company=record.company_id.id)
+                if record.user_id:
```

Comment:
maybe we should test if the fields exist too?

Reply:
resolved, thanks